scalafix-core/shared/src/main/scala/scalafix/internal/rule/ImplicitClassPrivateVal.scala

```diff
@@ -0,0 +1,33 @@
+package scalafix.internal.rule
+
+import scala.meta.quasiquotes.XtensionQuasiquoteInit
+import scala.meta.{Ctor, Defn, Mod, Template, Term}
+import scalafix.rule.RuleCtx
+import scalafix.{Patch, Rule}
+
+case object ImplicitClassPrivateVal extends Rule("ImplicitClassPrivateVal") {
+
+  override def description: String =
+    "Add private access modifier to val parameters of implicit value classes in order to prevent public access"
+
+  override def fix(ctx: RuleCtx): Patch = {
+    ctx.tree.collect {
+      case Defn.Class(
+          cMods,
+          _,
+          _,
+          Ctor.Primary(_, _, (Term.Param(pMods, _, _, _) :: Nil) :: Nil),
+          Template(_, init"AnyVal" :: Nil, _, _))
```

Comment:
we can lift the restriction that it cannot mix in universal classes, if you are feeling more ambitious 😄

Reply:
I think it's better to wait and see if people actually need that, because as you said, the error is more likely with implicit value classes. Speaking of, should I rename the rule to `ImplicitValueClassPrivateVal` ?

Reply:
> I think it's better to wait and see if people actually need that because as you said, the error is more likely to happen with implicit value classes.

I agree.

> Speaking of, should I rename the rule to ImplicitValueClassPrivateVal ?

Good question. How about `LeakingImplicitClassVal`? That gives us room to later support non-AnyVal in case there's demand. I think it's good to hint in the name what the problem is and the `Leaking*` prefix is consistent with other upcoming rules like `LeakingSealed`